DataViews: Improve UX around filter customization of bundled data views

Souptik2001 opened the pull request:
<!-- Thanks for contributing to Gutenberg! Please follow the Gutenberg Contributing Guidelines:
https://github.com/WordPress/gutenberg/blob/trunk/CONTRIBUTING.md -->

## What?
<!-- In a few words, what is the PR actually doing? -->

Improve UX around filter customization of bundled data views

## Why?
<!-- Why is this PR necessary? What problem is it solving? Reference any existing previous issue(s) or PR(s), but please add a short summary here, too -->

Addressed first three points of - #60468 

The UX of applying filters on a bundled view can be a little confusing.
For example we are on the "Scheduled" view -
- From there I can remove the "Scheduled" status filter.
- Also add other status filters.

But doing those we will still be on the "Scheduled" bundled view page. So, seeing that screen there is no relation between the filters applied and the view.

Also when we click on the "Reset" button when on any bundled view, it resets all the filter without respecting the view.

## How?
<!-- How is your PR addressing the issue at hand? What are the implementation details? -->

- Hide the filter option which is pre-applied by the view.
- Hide the "Reset" button when no other filter is applied other than the pre-applied filter by view.
- When "Reset" button is clicked reset the filter to the view pre-applied filter state.

## Testing Instructions
<!-- Please include step by step instructions on how to test this PR. -->
<!-- 1. Open a post or page. -->
<!-- 2. Insert a heading block. -->
<!-- 3. etc. -->

Check the screencast below or the points written -

- Go to site-editor.
- Open "Pages".
- On the "All pages" view, we will see all the filters are present.
- Now go to some other view like "Scheduled".
- We will see that the status filter disappears - neither in the apply filter dropdown or the applied filters chips.
- The RESET button will be also not visible by default.
- Now apply any other filter, like author. You will see the "Reset" button becomes visible.
- When you click on the "Reset" button all other filters gets reverted, except filters applied by the view.

### Testing Instructions for Keyboard
<!-- How can you test the changes by using the keyboard only? Please note, this is required for PRs that change the user interface (UI). This ensures the PR can be tested for any possible accessibility regressions. -->

## Screenshots or screencast <!-- if applicable -->


https://github.com/WordPress/gutenberg/assets/39427067/664686b2-f70a-47cf-adb7-388e693f8e35

Souptik2001 committed:
feat: Improve UX around filter customization of bundled data views

- Hide the filter option which is pre-applied by the view.
- Hide the "Reset" button when no other filter is applied other than the pre-applied filter by view.
- When "Reset" button is clicked reset the filter to the view pre-applied filter state.

Signed-off-by: Souptik Datta <souptikdatta2001@gmail.com>

jameskoster commented:
Thank you for the continued energy on these issues @Souptik2001 ❤️ 

This is working well for me on the Pages views, however I'm seeing errors on Templates and Patterns pages.

Souptik2001 committed:
fix: Add initalFilters data on templates view data also

Signed-off-by: Souptik Datta <souptikdatta2001@gmail.com>

Souptik2001 commented:
Ahhh it's a miss from my side. Sorry, I didn't properly test. 🙇 

Fixed it now. Now the templates should also behave the same way as we have implemented for pages!

Thanks @jameskoster! Please let me know if there are any other issues. 🙌

jameskoster commented:
Thanks for the update 👍 

We've most likely missed the boat for 6.6 now, so this will end up being a 6.7 consideration. That may mean delaying the merge a little, but that gives us time to get more feedback from @WordPress/gutenberg-design.

Souptik2001 commented:
Gotcha! Thanks @jameskoster! 🙇 

Will be keeping an eye here and will make any change as required.. 🙌

jasmussen commented:
@DaniGuardiola I pinged you on a related issue, would you be up for helping here as well? Thank you.

oandregal commented:
Hi @Souptik2001 thanks for preparing the PR and the patience in waiting for feedback.

I've taken a look at this and prepared an alternative at https://github.com/WordPress/gutenberg/pull/65264 Thought it'd be less intrusive that pushing to this PR. You still should be credited as a co-author there.

Souptik2001 commented:
Sure! thanks a lot @oandregal !

oandregal commented:
> I've taken a look at this and prepared an alternative at https://github.com/WordPress/gutenberg/pull/65264 

#65264 was closed, so I prepared a different alternative at https://github.com/WordPress/gutenberg/pull/65295

noisysocks commented:
Closing in favour of https://github.com/WordPress/gutenberg/pull/65295 which was merged.
